text/0092-wallet-registry.md

```diff
@@ -0,0 +1,116 @@
+
+1. Register a unique TON DNS name (eg. `tonkeeper.ton` or `tonhub.ton`).
+2. Advertise connection details as TON DNS records under this name as well as name, description and logo.
+3. Register the TON DNS name in the well-known wallet registry contract.
+
+### Discussion: alternative storage formats instead of TON DNS
+
+We realize that relying on TON DNS records for storing connection details, name and logo is an expansion of the [intended use](https://github.com/ton-blockchain/TEPs/blob/master/text/0081-dns-standard.md#dns-records) of TON DNS. If the community dislikes the expansion, alternative suggestions for discussion include:
+
+1. Storing the records on-chain as part of **wallet registry contract** storage, in an attribute dictionary per wallet provider similar to how [on-chain token metadata](https://github.com/ton-blockchain/TEPs/blob/master/text/0064-token-data-standard.md#content-representation) is stored.
+
+2. Storing the records off-chain as a JSON document on a URL hosted by each wallet provider, the **wallet registry contract** in this case will store the list of URLs.
+
+## App client flow
+
+Since this proposal is aiming to make it easier for app developers to build their clients, the flow from the dapp client point of view:
+
+1. We assume the dapp client is able to query data from the chain (this part is out of scope of this proposal and can rely on [toncenter](https://toncenter.com/api/v2/), directly on [ADNL via proxy](https://github.com/tonstack/wsadnlroxy) or any other way).
+2. Query the well-known wallet registry contract and get the list of TON DNS names for available wallet providers.
+3. Query the TON DNS records for each of these wallet providers to get each wallet's connection details, name, description and logo.
+4. Display the wallet provider list for the user as part of the "Connect Wallet" screen.
+5. The end-user chooses the wallet from the list that they're using, and if they don't have a supported wallet, can see an up-to-date list of wallets that they can download in order to use the app.
+
+Once the standard is approved, JavaScript client libraries that perform the above steps will be published in order to make integration into dapp clients as easy as possible.
+
+# Specification
+
+## Wallet registry contract
+
+The contract holds a list of TON DNS names of wallet providers in its persistent storage. Anyone can add a new TON DNS name to the list. To reduce spam, we propose to require a deposit of 1,000-10,0000 TON Coin for registration that will be held by the contract and returned if the wallet provider unregisters.
```

Comment:
The [ton value fluctuation](https://github.com/ton-blockchain/TEPs/pull/91#discussion_r967780500) issue is relevant here as well - if ton decreases in value temporarily, for example, it exposes the contract to a spam attack.